From 6ce0a7ba87c8670a5e7dd94f1d8751309c036a41 Mon Sep 17 00:00:00 2001
From: Zhantao Tang <zhantao.tang@windriver.com>
Date: Thu, 23 Mar 2023 11:38:25 +0800
Subject: [PATCH] ipc-shm: update to compatible with v5.15 kernel codes

Change the CONFIG_SOC_S32GEN1 to CONFIG_SOC_S32CC to compatible with
nxp-sdk-5.15 kernel.

Upstream-Status: Pending

Signed-off-by: Zhantao Tang <zhantao.tang@windriver.com>
---
 os/ipc-os.c                    | 2 +-
 sample/Makefile                | 2 +-
 sample/sample.c                | 2 +-
 sample_multi_instance/Makefile | 2 +-
 4 files changed, 4 insertions(+), 4 deletions(-)

diff --git a/os/ipc-os.c b/os/ipc-os.c
index 93d9e4e..dc31816 100644
--- a/os/ipc-os.c
+++ b/os/ipc-os.c
@@ -16,7 +16,7 @@
 #define DRIVER_VERSION	"0.1"
 
 /* Device tree MSCM node compatible property (search key) */
-#if defined(CONFIG_SOC_S32GEN1)
+#if defined(CONFIG_SOC_S32CC)
 	#define DT_INTC_NODE_COMP "nxp,s32cc-mscm"
 #elif defined(CONFIG_SOC_S32V234)
 	#define DT_INTC_NODE_COMP "fsl,s32v234-mscm"
diff --git a/sample/Makefile b/sample/Makefile
index a789e08..df1ca69 100644
--- a/sample/Makefile
+++ b/sample/Makefile
@@ -17,7 +17,7 @@ ifeq ($(POLLING),yes)
 ccflags-y += -DPOLLING
 endif
 
-ifeq ($(CONFIG_SOC_S32GEN1),y)
+ifeq ($(CONFIG_SOC_S32CC),y)
 ccflags-y += -DPLATFORM_FLAVOR_$(PLATFORM_FLAVOR)
 endif
 
diff --git a/sample/sample.c b/sample/sample.c
index ed92620..f4a59d4 100644
--- a/sample/sample.c
+++ b/sample/sample.c
@@ -23,7 +23,7 @@ MODULE_DESCRIPTION("NXP Shared Memory IPC Sample Application Module");
 MODULE_VERSION(MODULE_VER);
 
 /* IPC SHM configuration defines */
-#if defined(CONFIG_SOC_S32GEN1)
+#if defined(CONFIG_SOC_S32CC)
 #if defined(PLATFORM_FLAVOR_s32g3)
 	#define LOCAL_SHM_ADDR 0x34000000
 #elif defined(PLATFORM_FLAVOR_s32g2) || defined(PLATFORM_FLAVOR_s32r45)
diff --git a/sample_multi_instance/Makefile b/sample_multi_instance/Makefile
index cfe5d95..3f8bdbf 100644
--- a/sample_multi_instance/Makefile
+++ b/sample_multi_instance/Makefile
@@ -12,7 +12,7 @@ ifneq ($(KERNELRELEASE),)
 obj-m := $(MODULE_NAME).o
 $(MODULE_NAME)-y := sample_multi_instance.o
 
-ifeq ($(CONFIG_SOC_S32GEN1),y)
+ifeq ($(CONFIG_SOC_S32CC),y)
 ccflags-y += -DPLATFORM_FLAVOR_$(PLATFORM_FLAVOR)
 $(MODULE_NAME)-y += ipcf_Ip_Cfg_$(PLATFORM_FLAVOR).o
 endif
-- 
2.25.1

